lightning-background-processor/src/lib.rs

```diff
@@ -219,11 +219,18 @@ impl BackgroundProcessor {
 			let mut have_pruned = false;
 
 			loop {
-				peer_manager.process_events();
+				peer_manager.process_events(); // Note that this may block on ChannelManager's locking
 				channel_manager.process_pending_events(&event_handler);
 				chain_monitor.process_pending_events(&event_handler);
+
+				// We wait up to 100ms, but on a mobile device without a ton of activity, if we get
+				// put in the background and paused, this is likely where it will happen, causing
+				// the wait time to be substantially longer.
```

Comment:
Consider consolidating this comment and the one later talking about iOS putting an app in the background and desktops being asleep.